### PR TITLE
Add the missing type hints to our interfaces and abstract classes

### PR DESCRIPTION
**Pull request by leofeyer**

### CODE REVIEW

File: core-bundle/src/Image/ImageFactoryInterface.php [hunk 1]
```diff
@@ -25,17 +25,13 @@ interface ImageFactoryInterface
      * @param string|ImageInterface              $path    The absolute path to the source image or an Image object
      * @param int|array|ResizeConfiguration|null $size    An image size ID, an array with width, height and resize mode or a ResizeConfiguration object
      * @param string|ResizeOptions|null          $options The target path as string or a ResizeOptions object
-     *
-     * @return ImageInterface
      */
-    public function create($path, $size = null, $options = null)/*: ImageInterface*/;
+    public function create(string|ImageInterface $path, int|array|ResizeConfiguration|null $size = null, string|ResizeOptions|null $options = null): ImageInterface;
```
Comment (ausi) [inline core-bundle/src/Image/ImageFactoryInterface.php]:
```suggestion
    public function create(string|ImageInterface $path, int|string|array|ResizeConfiguration|null $size = null, string|ResizeOptions|null $options = null): ImageInterface;
```

See https://github.com/contao/contao/blob/602bd1fe3db1211ec708749d5564207f73e9f3d1/core-bundle/src/Image/ImageFactory.php#L71 and https://github.com/contao/contao/blob/602bd1fe3db1211ec708749d5564207f73e9f3d1/core-bundle/src/Image/ImageFactory.php#L113

File: core-bundle/src/Image/PictureFactory.php [hunk 2]
```diff
@@ -61,7 +61,7 @@ public function __construct(PictureGeneratorInterface $pictureGenerator, ImageFa
         $this->imagineOptions = $imagineOptions;
     }
 
-    public function setDefaultDensities($densities): self
+    public function setDefaultDensities(string $densities): static
     {
         $this->defaultDensities = (string) $densities;
```
Comment (ausi) [inline core-bundle/src/Image/PictureFactory.php]:
```suggestion
        $this->defaultDensities = $densities;
```

**Review by ausi — DISMISSED** [review]

**Review by ausi — DISMISSED** [review]

File: core-bundle/src/Image/PictureFactoryInterface.php [hunk 3]
```diff
@@ -21,21 +21,11 @@ interface PictureFactoryInterface
...
      * Creates a Picture object.
-     *
-     * @param string|ImageInterface                      $path
-     * @param int|string|array|PictureConfiguration|null $size
-     * @param ResizeOptions|null                         $options
-     *
-     * @return PictureInterface
      */
-    public function create($path, $size = null/*, ResizeOptions $options = null*/)/*: PictureInterface*/;
+    public function create(string|ImageInterface $path, int|string|array|PictureConfiguration|null $size = null, ResizeOptions $options = null): PictureInterface;
```
Comment (ausi) [inline core-bundle/src/Image/PictureFactoryInterface.php:30]: `ImageResult` has to be adapted too ☺️ https://github.com/contao/contao/blob/bbba9e17ebb64c6b2d47e2e5d5bdc489a1c062d1/core-bundle/src/Image/Studio/ImageResult.php#L79-L99

Comment (leofeyer) [reply core-bundle/src/Image/PictureFactoryInterface.php:30]: Changed in c886d2cf006ece888a8ff4bb8e1e56791c8d3fd3.

**Review by ausi — DISMISSED** [review]

**Pull request merged**